guide/build-config.md

```diff
@@ -147,9 +147,9 @@ The build environment used for builds has a unique id, name and system image id.
 
 We allow the user to specify the build environment using either `environmentId`, `environmentName` or the `systemImageId`. `environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what has been traditionally used, but it is problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers. Moreover, an environment may become deprecated over time, thus the environmentId should be kept up-to-date, which is not optimal.
```

Comment:
What about:

We allow the user to specify the build environment using either `environmentId`, `environmentName` or the `systemImageId`.  
`environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what has been traditionally used, but it can be problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers. To overcome the portability issue, it is possible to specify the `systemImageId` instead, which is the same across different PNC servers. Both `environmentId` and `systemImageId` **preserve** the docker-image version of the environment, which means that after some time the version may become deprecated and users may need to update the value from time to time. (**NOTE**: `systemImageIds` with `:latest` suffix will **no longer be available** in PNC).  
The `environmentName` in contrast, is the same between different PNC servers and thus portability across different PNC servers is not a concern. However the big difference is that environments specified by `environmentName` are always using **the latest released docker image**. This is important because there will be changes from time to time to the build configurations in PNC, which will cause **rebuilds** of the affected components.

Reply:
I like it :)